app/src/main/java/org/schabi/newpipe/local/holder/LocalPlaylistStreamItemHolder.java

```diff
@@ -111,13 +116,19 @@ public void updateState(final LocalItem localItem,
 
         if (item.getProgressMillis() > 0 && item.getStreamEntity().getDuration() > 0) {
             itemProgressView.setMax((int) item.getStreamEntity().getDuration());
-            if (itemProgressView.getVisibility() == View.VISIBLE) {
-                itemProgressView.setProgressAnimated((int) TimeUnit.MILLISECONDS
-                        .toSeconds(item.getProgressMillis()));
+            if (DependentPreferenceHelper
+                    .getPositionsInListsEnabled(itemProgressView.getContext())) {
+                if (itemProgressView.getVisibility() == View.VISIBLE) {
+                    itemProgressView.setProgressAnimated((int) TimeUnit.MILLISECONDS
+                            .toSeconds(item.getProgressMillis()));
+                } else {
+                    itemProgressView.setProgress((int) TimeUnit.MILLISECONDS
+                            .toSeconds(item.getProgressMillis()));
+                    ViewUtils.animate(itemProgressView, true, 500);
+                }
             } else {
                 itemProgressView.setProgress((int) TimeUnit.MILLISECONDS
                         .toSeconds(item.getProgressMillis()));
```

Comment:
Uhm, why are you showing progress even if positions in lists are disabled?

Also, I think you should move the `getPositionsInListsEnabled` condition to the `if` on line 117